src/libsyntax/parse/parser.rs

```diff
@@ -2323,6 +2324,16 @@ impl<'a> Parser<'a> {
                         self.recover_stmt();
                     }
                 }
+                if self.check(&token::Comma) {
```

Comment:
Nit: `check` adds the token into the "expected tokens" set, `self.token == &token::Comma` should be enough here.